scaling/src/test/java/io/quarkus/ts/openshift/scaling/ScalingOpenShiftIT.java

```diff
@@ -0,0 +1,111 @@
+import java.util.concurrent.TimeUnit;
+
+import static io.restassured.RestAssured.given;
+import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@OpenShiftTest
+public class ScalingOpenShiftIT {
+
+    public static final String DC_NAME = "test-scaling";
+
+    @TestResource
+    private OpenShiftUtil openShiftUtil;
+
+    @AfterEach
+    public void scaleBack() {
+        openShiftUtil.scale(DC_NAME, 1);
+    }
+
+    @Test
+    public void scaleUpTest() {
+        given()
+                .when().get("/scaling")
+                .then()
+                .statusCode(OK.getStatusCode());
+
+        openShiftUtil.scale(DC_NAME, 2);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(2);
+
+        Set<String> uniqueIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        await().atMost(1, TimeUnit.MINUTES).untilAsserted(() -> {
+            String uniqueId =
+                    given()
+                            .when().get("/scaling")
+                            .then()
+                            .statusCode(OK.getStatusCode())
+                            .extract().asString();
+            uniqueIds.add(uniqueId);
+
+            assertThat(uniqueIds).hasSize(2);
+        });
+    }
+
+    @Test
+    public void scaleDownTest() {
+        openShiftUtil.scale(DC_NAME, 2);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(2);
+
+        Set<String> uniqueIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        await().atMost(1, TimeUnit.MINUTES).untilAsserted(() -> {
+            String uniqueId =
+                    given()
+                            .when().get("/scaling")
+                            .then()
+                            .statusCode(OK.getStatusCode())
+                            .extract().asString();
+            uniqueIds.add(uniqueId);
+
+            assertThat(uniqueIds).hasSize(2);
+        });
+
+        openShiftUtil.scale(DC_NAME, 1);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(1);
+
+        uniqueIds.clear();
+        for (int i = 0; i < 100; i++) {
+            String uniqueId =
+                    given()
+                            .when().get("/scaling")
+                            .then()
+                            .statusCode(OK.getStatusCode())
+                            .extract().asString();
+            uniqueIds.add(uniqueId);
+        }
+
+        assertThat(uniqueIds).hasSize(1);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(1);
+    }
+
+    @Test
+    public void scaleToZero() {
+        openShiftUtil.scale(DC_NAME, 0);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(0);
+
+        for (int i = 0; i < 100; i++) {
```

Comment:
Why so many requests?

Reply:
It's an arbitrary amount, if you think it's too much, I guess I can reduce it.